tests/swarm-test.js

```diff
@@ -38,9 +38,16 @@ beforeEach(function (done) {
 })
 
 afterEach(function (done) {
-  swarmA.closeListener()
-  swarmB.closeListener()
-  done()
+  // This should be 2, but for some reason
+  // that will fail in most of the tests
```

Comment:
I see. When one of the swarm closes, the other swarm will get a ECONRESET on the socket and close itself. That should why the second `closeListener` is failing and not properly returning.